pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/windowing/WindowManager.java

```diff
@@ -105,7 +105,9 @@ public void add(T event, long ts, Record<?> record) {
     public void add(Event<T> windowEvent) {
         // watermark events are not added to the queue.
         if (windowEvent.isWatermark()) {
-            log.debug(String.format("Got watermark event with ts %d", windowEvent.getTimestamp()));
+            if (log.isDebugEnabled()) {
+                log.debug(String.format("Got watermark event with ts %d", windowEvent.getTimestamp()));
```

Comment:
No need to use String.format, just 
log.debug("Got watermark event with ts {}", windowEvent.getTimestamp());

Reply:
yes , I will delete